src/cryptsetup/cryptsetup-tpm2.c

```diff
@@ -64,7 +119,58 @@ int acquire_tpm2_key(
                 blob = loaded_blob;
         }
 
-        return tpm2_unseal(device, pcr_mask, pcr_bank, primary_alg, blob, blob_size, policy_hash, policy_hash_size, NULL, ret_decrypted_key, ret_decrypted_key_size);
+        if (flags & TPM2_FLAGS_USE_PIN) {
+                for (int i = 5;; i--) {
+                        _cleanup_(erase_and_freep) char *pin_str = NULL;
+
+                        if (i <= 0)
+                                return log_error_errno(SYNTHETIC_ERRNO(EPERM), "Failed to unlock, giving up!");
+
+                        r = get_pin(&pin_str);
+                        if (r == -EPERM)
+                                continue;
+                        if (r < 0)
+                                return r;
+
+                        r = tpm2_unseal(
+                                device,
```

Comment:
our usual coding style suggests to indent line broken parameter lists to 16ch, not 8ch, to make them visually distinct from code block indentation

Reply:
OK. Looks like clang-format won't do that?

Reply:
By all means, add a patch to this PR that adds it if we are missing that.

We are currently not using clang-format automatically, since it couldn't really format things for us the way we wanted. But I am sure it's improving, so we might have missed some settings.

ideally, one day it's good enough to be something we can do automatically and enforce, at which point we should hopefully notice mssing tweaks like that more quickly.